### DISCUSSION TITLE
How create g.dart ?

### DISCUSSION THREAD

**Discussion by kindKmj2** [isar/isar#1536] [Help]
this is my code 

```dart

import 'package:isar/isar.dart';

part 'sample.g.dart';

@Collection()
@Name('sample')
class Sample{


  Id id = Isar.autoIncrement;
  final String name;

  Sample(this.id, this.name);
}
```

i use command 
''' dart
flutter pub run build_runner build
'''

this my pud.dev code 

```dart
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.6
  flutter_bloc: ^8.1.3
  sealed_flutter_bloc: ^8.0.1
  dio: ^5.4.0
  get_it: ^7.6.4
  flutter_screenutil: ^5.9.0
  kakao_flutter_sdk: ^1.8.0
  flutter_svg: ^2.0.9
  gap: ^3.0.1
  flutter_inappwebview: ^6.0.0
  lottie: ^2.7.0
  isar: ^3.1.0+1
  isar_flutter_libs: ^3.1.0+1
  path_provider: ^2.1.2

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0
  isar: ^3.1.0+1
  build_runner: ^2.4.7
  json_serializable: ^6.7.1
```

and  log here 

```
Deprecated. Use `dart run` instead.
[INFO] Generating build script completed, took 135ms
[INFO] Reading cached asset graph completed, took 158ms
[INFO] Checking for updates since last build completed, took 600ms
[INFO] Running build completed, took 8.3s
[INFO] Caching finalized dependency graph completed, took 114ms
[INFO] Succeeded after 8.4s with 429 outputs (864 actions)

```

i don`t understand 
why not create g.dart file?

plz help me

**Comment by kindKmj2** [comment]
my mistake 
 isar_generator: ^3.1.0+1 
 update pub.dev this package 
and solved

**Comment by jonasermert** [comment]
Hello, make sure there are no errors in your sample.dart file. Any syntax or logical errors might prevent the generation of the g.dart file. Ensure that you have the necessary annotations in your code. In your case, the @Collection and @Name annotations seem correct. Double-check the Isar documentation to ensure you are using the annotations correctly. After running the flutter pub run build_runner build command, carefully check the console output for any error messages or warnings related to the code generation process. Ensure that your pubspec.yaml file has the correct configuration for the build_runner and json_serializable. You have these dependencies in your pubspec.yaml, but make sure they are set up correctly. Sometimes, cached files might cause issues. Try cleaning your project and rebuilding it by running the following commands:

flutter clean
flutter pub get
flutter pub run build_runner build

Ensure that your Flutter SDK is up-to-date. You can check for the latest version using flutter upgrade.
